codegen/src/main/scala/bsp/codegen/docs/TypescriptRenderer.scala

```diff
@@ -150,18 +150,23 @@ class TypescriptRenderer {
     case PTimestamp => "number"
   }
 
-  def renderDocumentation(hints: List[Hint]): Lines = hints
-    .collectFold { case Documentation(string) =>
-      val lines = string.split(System.lineSeparator())
-      if (lines.nonEmpty) {
-        lines(0) = "/** " + lines(0)
-        val lastIndex = lines.length - 1
-        for (i <- 1 to lastIndex) {
-          lines(i) = " * " + lines(i)
-        }
-        lines(lastIndex) = lines(lastIndex) + " */"
+  def renderComments(string: String): Lines = {
+    val lines = string.split(System.lineSeparator())
+    if (lines.nonEmpty) {
+      lines(0) = "/** " + lines(0)
+      val lastIndex = lines.length - 1
+      for (i <- 1 to lastIndex) {
+        lines(i) = " * " + lines(i)
       }
-      Lines(lines.toList)
+      lines(lastIndex) = lines(lastIndex) + " */"
+    }
+    Lines(lines.toList)
+  }
+
+  def renderDocumentation(hints: List[Hint]): Lines = hints
+    .collectFold {
+      case Documentation(string) => renderComments(string)
+      case Deprecated(message)   => renderComments(s"@deprecated $message")
```

Comment:
That's incorrect, as now if there's both a Documentation hint and a Deprecated hint two comment blocks will be rendered